hs/src/Reach/Simulator/Core.hs

```diff
@@ -632,27 +646,60 @@ instance Interp LLStep where
       _ <- interp stmt
       interp step
     LLS_Stop _at -> return V_Null
-    LLS_ToConsensus _at _lct tc_send (DLRecv {..}) _tc_mtime -> do
+    LLS_ToConsensus _at _lct tc_send recv tc_mtime -> do
       (g, l) <- getState
       let mid = e_nmsgid g
       incrMessageId
-      v <- suspend $ PS_Suspend (A_TieBreak mid $ M.keys $ M.mapKeys bunpack tc_send)
+      let sends = M.mapKeys bunpack tc_send
+      v <- suspend $ PS_Suspend (A_TieBreak mid $ M.keys sends)
       case v of
         V_UInt n -> do
           let locals = e_locals l
-          let lclst = l_who $ saferMapRef "LLS_ToConsensus" $ M.lookup (fromIntegral n) locals
+          let lclsv = saferMapRef "LLS_ToConsensus" $ M.lookup (fromIntegral n) locals
+          let lclst = l_who $ lclsv
           case lclst of
             Nothing -> impossible "expected participant id, received consensus id"
             Just part -> do
-              let m = saferMapRef "LLS_ToConsensus1" $ M.lookup (bpack part) tc_send
-              case m of
-                DLSend {..} -> do
-                  ds_msg' <- mapM interp ds_msg
-                  _ <- zipWithM addToStore dr_msg ds_msg'
-                  _ <- zipWithM (addToRecord $ fromIntegral n) dr_msg ds_msg'
-                  interp $ dr_k
+              let dls = saferMapRef "LLS_ToConsensus1" $ M.lookup part sends
+              let f t n' step = case (t < n') of
+                    True -> consensusBody n g l dls recv
+                    False -> interp step
+              case tc_mtime of
+                Just (dltimearg, step) -> case dltimearg of
+                  Left dlarg -> do
+                    n' <- vUInt <$> interp dlarg
+                    let t = e_nwtime g
+                    f t n' step
+                  Right dlarg -> do
+                    n' <- vUInt <$> interp dlarg
+                    let t = e_nwtime g
+                    f t n' step
```

Comment:
This doesn't make any sense. The only way you can run the timeout is if the time advanced. But you have this logic inside of the "tie breaking", but really there would be another action called "Run Timeout" but this could only be used (if it would only not error) if the time is past